BraveRewardsUI/Settings/Tips/Details/TipsDetailViewController.swift

```diff
@@ -182,3 +247,24 @@ extension TipsDetailViewController {
     }
   }
 }
+
+extension TipsDetailViewController {
+  fileprivate func getTipsThisMonth() -> BalanceReportInfo {
+    let components = Calendar(identifier: .gregorian).dateComponents([.month, .year], from: Date())
+    let month = components.month ?? -1
+    let year = components.year ?? 2019
+    return state.ledger.balanceReport(for: ActivityMonth(rawValue: month) ?? .any, year: Int32(year))
+  }
+  
+  fileprivate func setFavicon(url: String, indexPath: IndexPath) {
```

Comment:
Using index path here is not a good way of updating the cell, since index paths could change if the table is reloaded at any point.

Better to pass in the tip, which then give you an identifier to update based on where the tip is in the list rather than where it was when you started the request.

Reply:
Instead of tip or indexpath I updated to using url and id so that we dont strongly capture Tip.

Reply:
> Instead of tip or indexpath I updated to using url and id so that we dont strongly capture Tip.
I don't understand what you mean… Just change `[weak self]` to `[weak self, weak tip]` or whatever the syntax is

Reply:
Agreed I just took the identifier as thats what we actually need. Saves us the check for weak tip.